apps/backend/src/datasources/postgres/ReviewDataSource.ts

```diff
@@ -336,4 +336,64 @@ export default class PostgresReviewDataSource implements ReviewDataSource {
         return reviews.map((review) => createReviewObject(review));
       });
   }
+
+  /*Brief explanation of the query used in getAllUsersReviews
+    This query gets executed when a reviewer wishes to see all the proposals
+    (proposals where they are assigned as a reviewer and where they are not) 
+    in a FAP. Logged in reviewer's id will be returned as the reviewer against 
+    a proposal if the proposal is being reviewed by him and others. Reviewers
+    other than him will be ignored for that proposal.
+  */
+  async getAllUsersReviews(
+    fapIds: number[],
+    userId?: number,
+    callId?: number,
+    instrumentId?: number,
+    status?: ReviewStatus
+  ): Promise<Review[]> {
+    return database
+      .select('fapReviewsTemp.*')
+      .from(
+        database
+          .select('fap_reviews.*')
+          .from('fap_reviews')
+          .modify((qb) => {
+            if (callId) {
+              qb.join('proposals', {
+                'proposals.proposal_pk': 'fap_reviews.proposal_pk',
+              });
+              qb.where('proposals.call_id', callId);
+            }
+            if (instrumentId) {
+              qb.join('instrument_has_proposals', {
+                'instrument_has_proposals.proposal_pk':
+                  'fap_reviews.proposal_pk',
+              });
+              qb.where('instrument_has_proposals.instrument_id', instrumentId);
+            }
+
+            if (status) {
+              qb.where('fap_reviews.status', status);
+            }
+          })
+          .whereIn('fap_id', fapIds)
+          .as('fapReviewsTemp')
+      )
+      .modify((query) =>
+        query
+          .where('user_id', userId)

```

Comment:
Thank you for the review @TCMeldrum but the problem is not with the where clause, its because null check for status was removed. Added back the null check now its showing properly. Please could you check again and let me know.